controllers/numaresourcesoperator_controller.go

```diff
@@ -771,6 +773,15 @@ func daemonsetUpdater(mcpName string, gdm *rtestate.GeneratedDesiredManifest) er
 		klog.V(5).InfoS("DaemonSet update: cannot update config", "mcp", mcpName, "daemonset", gdm.DaemonSet.Name, "error", err)
 		return err
 	}
+	if gdm.ClusterPlatform != platform.Kubernetes {
+		if instance.IsCustomPolicyEnabled() && gdm.ClusterPlatform == platform.OpenShift {
```

Comment:
I'd rather move the flag into the `GeneratedDesiredManifest` struct

Reply:
Done.